phongo_compat.h

```diff
@@ -92,9 +92,17 @@
 #  define ARG_UNUSED
 #endif
 
-#if SIZEOF_LONG == 4
+#if PHP_VERSION_ID < 70000
+typedef long zend_long;
+# define SIZEOF_ZEND_LONG SIZEOF_LONG
+#endif
+
+#if SIZEOF_ZEND_LONG == 8
+# define ADD_INDEX_INT64(zval, index, value) add_index_long(zval, index, value)
+# define ADD_ASSOC_INT64(zval, key, value) add_assoc_long(zval, key, value)
+#elif SIZEOF_ZEND_LONG == 4
 # define ADD_INDEX_INT64(zval, index, value) \
-	if (value > LONG_MAX || value < LONG_MIN) { \
+	if (value > INT32_MAX || value < INT32_MIN) { \
```

Comment:
I found that these macros were inconsistent with those in `bson.c`, which used `INT_MAX` and `INT_MIN`. The `INT32` constants are more explicit.